CoreAlgorithmPlugins/src/au/gov/asd/tac/constellation/plugins/algorithms/clustering/ktruss/KTruss.java

```diff
@@ -259,8 +259,8 @@ private static int getComponents(GraphWriteMethods graph, BitSet links, Map<Inte
         for (int linkPosition = links.nextSetBit(0); linkPosition >= 0; linkPosition = links.nextSetBit(linkPosition + 1)) {
             getComponentsHopper(graph, links, nodeToComponent, linkToComponent, componentTree, currentComponentNum, linkPosition);
             int componentCounter = 0;
-            for (Integer key : nodeToComponent.keySet()) {
-                if (nodeToComponent.get(key) == currentComponentNum) {
+            for (Map.Entry<Integer, Integer> entry : nodeToComponent.entrySet()) {
```

Comment:
Perhaps make the Map.Entry references final?

Reply:
Done